include/tvm/runtime/name_mangling.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file tvm/runtime/name_mangling.h
+ * \brief Utility functions for name mangling which are used
```

Comment:
Given the name transforms are fairly standalone, can we move them all rather than having some here and some in the original file?

Reply:
I moved this one since it is shared between runtime library and tvm library. I think it's better to keep the rest in src/backend since they are not used in runtime currently. wdyt?

Reply:
I don't see much harm moving them all across, they're small, self contained and it's likely if you need one you need others. Either way, it'd be good not to have `name_mangling.h` and `name_transforms.h` as it confuses people, if you don't want to move it together could we at least name them consistently? "Mangle" is also an unnecessarily aggressive word, we should probably not be using it in our codebase.

Reply:
Makes sense, I'll rename it to `name_transforms`